src/transformers/models/whisper/modeling_whisper.py

```diff
@@ -1608,6 +1614,18 @@ def generate(
         if len(forced_decoder_ids) > 0:
             generation_config.forced_decoder_ids = forced_decoder_ids
 
+        if initial_prompt_ids is not None:
+            if "decoder_start_token_id":
+                raise ValueError(
+                    "When specifying `initial_prompt_ids`, you cannot also specify `decoder_start_token_id` as it gets overwritten."
+                )
```

Comment:
Replacing this with an actual conditional now. Any idea how the model test I added passed with this?